packages/azure-http-specs/specs/azure/core/page/main.tsp

```diff
@@ -206,3 +206,54 @@ interface TwoModelsAsPageItem {
   @route("/second-item")
   listSecondItem is global.Azure.Core.Foundations.Operation<{}, global.Azure.Core.Page<SecondItem>>;
 }
+
+model IncludePendingOptions {
+  @query
+  includePending?: boolean;
+}
+
+@global.Azure.Core.pagedResult
+model ParameterizedNextLinkPagingResult {
+  @global.Azure.Core.items
+  values: User[];
+
+  @nextLink
+  nextLink: global.Azure.Core.Legacy.parameterizedNextLink<[IncludePendingOptions.includePending]>;
+}
+
+@scenario
+@doc("List with parameterized next link that re-injects parameters.")
+@route("/with-parameterized-next-link")
+@scenarioDoc("""
+    This scenario tests the Azure.Core.Legacy.parameterizedNextLink decorator which ensures original request
+    parameters are maintained in next link URLs.
+    
+    Expected query parameters on initial request:
+    - includePending=true
+    - select=name
+  
+    Expected query parameters on next link request. Note: the SDK will need to re-inject this parameter:
+    - includePending=true
```

Comment:
`select=name` shall also be included in second call.

Reply:
according to the `Azure.Core.Legacy.parameterizedNextLink` definition, only `includePending` should be re-inject. whether `select` is included is depend on service returned `nextLink`.

Reply:
i'll go ahead and add it back to avoid confusion